CBinding/CMake/CMakeProject.cs

```diff
@@ -0,0 +1,349 @@
+						var t = getFileAndLine (line, " at ");
+						lineNumber = t.Item1;
+						fileName = t.Item2;
+					} else {
+						string[] error = line.Split (':');
+						if (!String.IsNullOrEmpty (error.ElementAtOrDefault (1))) {
+							sb.Append (error [1]);
+						}
+					}
+				} else {
+					sb.Append (line);
+				}
+			}
+			
+			return results;
+		}
+		
+		public void RemoveTarget (string targetName)
+		{
+			fileFormat.RemoveTarget (targetName);
+		}
+		
+		protected override string OnGetBaseDirectory ()
+		{
+			return file.ParentDirectory.ToString();
+		}
+		
+		protected override string OnGetName ()
+		{
+			return this.name;
+		}
+		
+		public void LoadFrom (FilePath file)
+		{
+			this.file = file;
+			FileName = file;
+			CMakeFileFormat fileFormat = new CMakeFileFormat (file, this);
+			fileFormat.Parse ();
+			name = fileFormat.ProjectName;
+			this.fileFormat = fileFormat;
+		}
+		
+		protected override Task OnSave (ProgressMonitor monitor)
+		{
+			return Task.Factory.StartNew (() => {
+				fileFormat.SaveAll ();
+			});
+		}
+		
+		protected override IEnumerable<WorkspaceObject> OnGetChildren ()
+		{
+			foreach (CMakeTarget target in fileFormat.Targets.Values)
+				target.ParentObject = this;
+			return fileFormat.Targets.Values.ToList ();
+		}
+		
+		/*
+		List<FilePath> files = new List<FilePath> ();
+		protected override IEnumerable<FilePath> OnGetItemFiles (bool includeReferencedFiles)
+		{
+			files.Clear ();
+			foreach (CMakeTarget target in fileFormat.Targets.Values) {
+				files = files.Concat (target.GetFiles ()).ToList ();
+			}
+			return files;
+		}
+		
+		CMakeTarget getTarget (FilePath fileName)
+		{
+			string filename = fileName.FileName;
+			foreach (CMakeTarget target in fileFormat.Targets.Values) {
+				foreach (string file in target.Files.Keys) {
+					if (file.EndsWith (filename, StringComparison.OrdinalIgnoreCase))
+						return target;
+				}
+			}
+			return null;
+		}
+		
+		void AddFile (FilePath fileName, string targetName)
+		{
+			fileName = fileName.ToRelative (file.ParentDirectory);
+			foreach (string target in fileFormat.Targets.Keys) {
+				if (target.StartsWith (targetName, StringComparison.OrdinalIgnoreCase))
+					fileFormat.Targets [target].AddFile (fileName.ToString ());
+			}
+		}
+		
+		void RemoveFile (FilePath fileName)
+		{
+			fileName = fileName.ToRelative (file.ParentDirectory);
+			foreach (CMakeTarget target in fileFormat.Targets.Values) {
+				target.RemoveFile (fileName);
+			}
+		}
+		//*/
+		
+		protected override Task<BuildResult> OnBuild (ProgressMonitor monitor, ConfigurationSelector configuration, OperationContext operationContext)
+		{
+			return Task.Factory.StartNew (() => {
+				BuildResult results;
+				
+				if (!checkCMake ()) {
+					results = new BuildResult();
+					results.AddError ("CMake cannot be found.");
+					return results;
+				}
+				
+				FileService.CreateDirectory (Path.Combine (file.ParentDirectory.ToString (), outputDirectory));
```

Comment:
The ToString here is redundant, FilePath is implicitly castable to string.

Actually, you can go better using the Combine method on FilePath:

    FileService.CreateDirectory (file.ParentDirectory.Combine (outputDirectory));